File: frontend/src/app/shared/dto/MemberRequestDTO.ts
import {UserDTO} from './UserDTO';
import {NgoDTO} from './NgoDTO';
export enum MemberRequestStatus {
  PENDING,
  ACCEPTED,
  DECLINED
}
export class MemberRequestDTO {
  public id: number;
  public user: UserDTO;
  public ngo: NgoDTO;
  public status: MemberRequestStatus;
  public motivation: string;
}

File: source/src/main/java/ro/fii/licenta/api/dto/MemberRequestDTO.java
package ro.fii.licenta.api.dto;

public class MemberRequestDTO {

	private Long id;

	private UserDTO user;

	private NgoDTO ngo;

	private String motivation;

	private int status;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public UserDTO getUser() {
		return user;
	}

	public void setUser(UserDTO user) {
		this.user = user;
	}

	public NgoDTO getNgo() {
		return ngo;
	}

	public void setNgo(NgoDTO ngo) {
		this.ngo = ngo;
	}

	public String getMotivation() {
		return motivation;
	}

	public void setMotivation(String motivation) {
		this.motivation = motivation;
	}

	public int getStatus() {
		return status;
	}

	public void setStatus(int status) {
		this.status = status;
	}

}
